packages/utils/src/types.ts

```diff
@@ -0,0 +1,47 @@
+import type { QuestionDTO } from "@core/question.js";
+import {
+  LanguageCode,
+  CountryCode,
+  surportedLocales,
+  userGenders,
+  questionCategories,
+} from "./constants.js";
+
+type RemoveResult = { success: boolean };
+
+type PaginationInput = { limit?: number; cursor?: string };
+
+type InsertItemResult<T> = { item: T; nextItemId?: string };
+
+type SelectItem<TValue extends string = string> = {
+  text: string;
+  value: TValue;
+  selectItems?: SelectItem;
+};
+
+type SelectItemMetadata<TValue extends string = string> = {
+  [value in TValue]: SelectItem<TValue>;
+};
+
+type LocaleCode = `${LanguageCode}_${CountryCode}`;
+
+type SurportedLocale = (typeof surportedLocales)[number];
+
+type UserGender = (typeof userGenders)[number];
+
+export const aa: QuestionDTO = {};
```

Comment:
I import this "QuestionDTO" type from other file but my vscode shows it is any. I assign {} to aa but it have no type error

Reply:
This part should throw the type error because the {} is wrong